File: model/src/main/java/com/itransition/fanfictionbackend/model/Role.java
package com.itransition.fanfictionbackend.model;

import static javax.persistence.EnumType.STRING;
import com.itransition.fanfictionbackend.model.enums.ERole;
import lombok.Data;
import lombok.EqualsAndHashCode;

import javax.persistence.Entity;
import javax.persistence.Enumerated;
import javax.persistence.Table;

@Data
@Entity
@Table(name = "role")
@EqualsAndHashCode(callSuper = true)
public class Role extends WithLongId {

    @Enumerated(STRING)
    private ERole name;
    private Boolean signUpRole;
}

File: src/main/kotlin/com/itransition/fanfictionbackend/model/Role.kt
package com.itransition.fanfictionbackend.model

import com.itransition.fanfictionbackend.model.enums.ERole
import javax.persistence.Entity
import javax.persistence.EnumType.STRING
import javax.persistence.Enumerated
import javax.persistence.Table

@Entity
@Table(name = "role")
class Role : WithLongId() {

    @Enumerated(STRING)
    var name: ERole? = null

    var signUpRole: Boolean = false
}
